regression/contracts/assigns_enforce_structs_07/main.c

```diff
@@ -28,6 +28,7 @@ int main()
   struct pair *p = nondet_bool() ? malloc(sizeof(*p)) : NULL;
   f1(p);
   struct pair_of_pairs *pp = malloc(sizeof(*pp));
+  pp->p = nondet_bool() ? malloc(sizeof(*(pp->p))) : NULL;
```

Comment:
Shall we use `--malloc-may-fail` and `--malloc-fail-null` in this test case instead?

Reply:
Good idea. Done.